File: services/influencer_crm/schemas/ops.py
"""Pydantic schemas for the /ops/health endpoint."""
from __future__ import annotations
from typing import Optional

from datetime import datetime

from pydantic import BaseModel


class EtlLastRun(BaseModel):
    started_at: Optional[datetime] = None
    status: Optional[str] = None
    duration_ms: Optional[int] = None
    error_message: Optional[str] = None


class EtlCounts(BaseModel):
    success: int = 0
    failed: int = 0
    running: int = 0
    stale_running: int = 0  # running > 1h, treated as stuck


class CronJobInfo(BaseModel):
    jobname: str
    schedule: str
    active: bool


class RetentionCounts(BaseModel):
    audit_log_eligible_for_delete: int = 0
    snapshots_eligible_for_delete: int = 0


class OpsHealth(BaseModel):
    etl_last_run: EtlLastRun
    etl_last_24h: EtlCounts
    mv_age_seconds: Optional[int] = None
    retention: RetentionCounts
    cron_jobs: list[CronJobInfo]

File: services/influencer_crm_ui/src/api/ops.ts
import { api } from '@/lib/api';

// Mirrors services/influencer_crm/schemas/ops.py (OpsHealth).
// Optional fields use `| null` to match Pydantic's `T | None = None`.
export interface EtlLastRun {
  started_at: string | null;
  status: string | null;
  duration_ms: number | null;
  error_message: string | null;
}

export interface EtlCounts {
  success: number;
  failed: number;
  running: number;
  stale_running: number;
}

export interface CronJobInfo {
  jobname: string;
  schedule: string;
  active: boolean;
}

export interface RetentionCounts {
  audit_log_eligible_for_delete: number;
  snapshots_eligible_for_delete: number;
}

export interface OpsHealth {
  etl_last_run: EtlLastRun;
  etl_last_24h: EtlCounts;
  mv_age_seconds: number | null;
  retention: RetentionCounts;
  cron_jobs: CronJobInfo[];
}

export function getOpsHealth(): Promise<OpsHealth> {
  return api.get<OpsHealth>('/ops/health');
}
